scipy/io/wavfile.py

```diff
@@ -419,21 +419,42 @@ def _read_data_chunk(fid, format_tag, channels, bit_depth, is_big_endian,
 
     # Number of bytes per sample (sample container size)
     bytes_per_sample = block_align // channels
-    if bit_depth == 8:
-        dtype = 'u1'
-    else:
-        if format_tag == WAVE_FORMAT.PCM:
+
+    if format_tag == WAVE_FORMAT.PCM:
+        if 1 <= bit_depth <= 8 or bytes_per_sample == 3:
+            dtype = 'u1'
+        elif bit_depth <= 32:
             dtype = f'{fmt}i{bytes_per_sample}'
         else:
+            raise ValueError("Unsupported bit depth: the wav file "
+                             f"has {bit_depth}-bit data.")
+    elif format_tag == WAVE_FORMAT.IEEE_FLOAT:
+        if bit_depth in {16, 32, 64}:
             dtype = f'{fmt}f{bytes_per_sample}'
+        else:
+            raise ValueError("Unsupported bit depth: the wav file "
+                             f"has {bit_depth}-bit data.")
+    else:
+        # This should have been caught by _read_fmt_chunk already
+        raise ValueError(f"Unknown wave file format")
```

Comment:
Missing a {format_tag} here.

Reply:
Yeah, that checking is already covered inside the read fmt chunk function, so this should never be reached, so I wasn't sure what to do here.  Duplicate the same code from the read fmt function?

```
        try:
            format_name = WAVE_FORMAT(format_tag).name
        except ValueError:
            format_name = f'{format_tag:#06x}'
        raise ValueError(f"Unknown wave file format: {format_name}. Supported "
                         "formats: " +
                         ', '.join(x.name for x in KNOWN_WAVE_FORMATS))
```

Reply:
maybe a little helper `_bad_format(format_name)` so that there is less code dup?